src/lib.rs

```diff
@@ -135,7 +134,7 @@ impl<F: PrimeField, PC: PolynomialCommitment<F, DensePolynomial<F>>, D: Digest>
         };
 
         let index_pk = IndexProverKey {
-            index,
+            index: index.clone(),
```

Comment:
Why the clone here?

Reply:
False alarm in Clion.
Let me remove this clone.